### ISSUE TITLE
startServer.sh high cpu usage

### ISSUE THREAD

**Issue by csegedicsaba** [oracle/weblogic-kubernetes-operator#2655]
After upgrading Openshift 4.6 to 4.8 (kubernetes 1.19 to 1.21) CPU usage increased.

After starting pods the cpu usage is increasing from 0.1 to 0.3 in ca. 5 days. After 5 days, the usage fall back to 0.1 and increases very slowly.

The extra cpu usage is not in the java process, it is in startServer.sh.
Using strace we found that waitForShutdownMarker() in operator\src\main\resources\scripts\utils.sh is responsible for cpu usage.

```
function waitForShutdownMarker() {
  #
  # Wait forever.   Kubernetes will monitor this pod via liveness and readyness probes.
  #
  trace "Wait indefinitely so that the Kubernetes pod does not exit and try to restart"
  while true; do
    if [ -e ${SHUTDOWN_MARKER_FILE} ] ; then
      exit 0
    fi
    sleep 0.1
  done
}
```

Running only this small piece of script in a container with the same image the behavior is the same.

Are there any option to change this behavior without changing the operator and the weblogic base image?

**Comment by ankedia** [comment]
@csegedicsaba - Will it be possible to try changing the script to increase the sleep value from `0.1` seconds to `0.5` or `1` in your test pod and see if it helps the issue? We can make it a tunable if that resolves the high cpu usage issue.

Do you configure the CPU requests and limits for your pod?  You can also try configuring the CPU affinity for the Pod using CPU management policy of kubelet. See - [Configuring CPU affinity](https://oracle.github.io/weblogic-kubernetes-operator/faq/resource-settings/#configuring-cpu-affinity) and [the Kubernetes documentation](https://kubernetes.io/docs/tasks/administer-cluster/cpu-management-policies/).

**Comment by csegedicsaba** [comment]
Increasing the sleep value to 1 second  the cpu usage is not significant.

It would be great to have possibility to override the default sleep value to solve this strange issue.

**Comment by ankedia** [comment]
Thanks @csegedicsaba . I have created a JIRA to increase the default sleep value to 1 second and allow overriding of the default value using an environment variable defined in the domain specifications. We will let you know once it is released.

**Comment by rjeberhard** [comment]
Resolved in 3.3.8.
